### PR TITLE
Single document UI

### PR DESCRIPTION
**Pull request by afshin**
This PR:
* Rationalizes the ranking of items in the left sidebar. The default rank is `500`, and all the JupyterLab components that add themselves to the left sidebar pick increments of `100`.
* Enables `multiple-document` and `single-document` modes for the main application dock panel.
* Creates a tab manager in the left sidebar to show all open documents regardless of what mode the application is in, allowing users to switch documents even in `single-document` mode.
* The application mode can be toggled with:
  * A global keyboard shortcut, currently: `Accel Shift Enter`
  * A command in the command palette (its label toggles based on current mode)
* The application mode is part of the state that gets restored upon refresh.

cc: @jasongrout @ellisonbg @cameronoelsen

### CODE REVIEW

**Comment by jasongrout** [comment]
CC @ivanov

File: packages/application/src/shell.ts [hunk 1]
```diff
@@ -174,7 +156,7 @@ class ApplicationShell extends Widget {
   }
 
   /**
-   * Activate a widget in it's area.
+   * Activate a widget in its area.
```
Comment (jasongrout) [inline packages/application/src/shell.ts]: nice!

**Comment by afshin** [comment]
### A notebook in SDI mode
notebook sdi

### A terminal in SDI mode
terminal sdi

### The mode toggling command
command

**Comment by afshin** [comment]
The styles still need some tweaking and the behaviors have not been implemented yet, but the semantics of this UI are: there's a "Tabs" list in the left sidebar and it allows you to set the current document even in single-document mode:

screen shot 2017-04-11 at 9 24 44 pm

cc: @ellisonbg @jasongrout @cameronoelsen @ivanov if you have any feedback, I'd appreciate it!

**Comment by willingc** [comment]
Thanks @afshin! I'm looking forward to trying this out :smile:

**Pull request merged**